Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.cs

```diff
@@ -1263,6 +1288,50 @@ private static void OnIsValidPropertyChanged(DependencyObject d, DependencyPrope
             }
         }
 
+        /// <summary>
+        /// Gets or sets the threshold range that governs when the DataGrid class will begin to prefetch more items.
+        /// </summary>
+        /// <returns>
+        /// The loading threshold, in terms of pages.
+        /// </returns>
+        public double IncrementalLoadingThreshold
+        {
+            get { return (double)GetValue(IncrementalLoadingThresholdProperty); }
+            set { SetValue(IncrementalLoadingThresholdProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="IncrementalLoadingThreshold"/> dependency property
+        /// </summary>
+        public static readonly DependencyProperty IncrementalLoadingThresholdProperty =
+            DependencyProperty.Register(
+                "IncrementalLoadingThreshold",
+                typeof(double),
+                typeof(DataGrid),
+                new PropertyMetadata(DATAGRID_defaultIncrementalLoadingThreshold));
```

Comment:
There must be an OnIncrementalLoadingThresholdPropertyChanged method in which you call LoadMoreDataFromIncrementalItemsSource() when the new value is larger than the old value.

Also look what OnMaxColumnWidthPropertyChanged does:  OnIncrementalLoadingThresholdPropertyChanged should also throw when e.NewValue is double.NaN or is < 0, and restore e.OldValue the same way.

Reply:
Actually Filip, for now ignore the "in which you call LoadMoreDataFromIncrementalItemsSource() when the new value is larger than the old value" part of my comment above. 
I want to discuss this with Harini first. We may want to stick to the basic principle that incremental loading is *only* kicked in when the scrolled off height decreases. Thanks.

Reply:
@harinikmsft So the conclusion is that it should not be called within OnIncrementalLoadingThresholdPropertyChanged, if I understand you correctly?